packages/angular_devkit/build_webpack/src/utils/webpack-file-system-host-adapter.ts

```diff
@@ -0,0 +1,154 @@
+    const p = normalize('/' + path);
+    const result = this._host.stat(p);
+
+    if (result === null) {
+      const o = this._host.exists(p).pipe(
+        switchMap(exists => {
+          if (!exists) {
+            throw new FileDoesNotExistException(p);
+          }
+
+          return this._host.isDirectory(p).pipe(
+            mergeMap(isDirectory => {
+              return (isDirectory ? of(0) : this._host.read(p).pipe(
+                map(content => content.byteLength),
+              )).pipe(
+                map(size => [isDirectory, size]),
+              );
+            }),
+          );
+        }),
+        map(([isDirectory, size]) => {
+          return {
+            isFile() { return !isDirectory; },
+            isDirectory() { return isDirectory; },
+            size,
+            atime: new Date(),
+            mtime: new Date(),
+            ctime: new Date(),
+            birthtime: new Date(),
+          };
+        }),
+      );
+
+      this._doHostCall(o, callback);
+    } else {
+      this._doHostCall(result, callback);
+    }
+  }
+
+  readdir(path: string, callback: Callback<string[]>): void {
+    return this._doHostCall(this._host.list(normalize('/' + path)), callback);
+  }
+
+  readFile(path: string, callback: Callback<Buffer>): void {
+    const o = this._host.read(normalize('/' + path)).pipe(
+      map(content => new Buffer(content)),
+    );
+
+    return this._doHostCall(o, callback);
+  }
+
+  readJson(path: string, callback: Callback<JsonObject>): void {
+    const o = this._host.read(normalize('/' + path)).pipe(
+      map(content => JSON.parse(virtualFs.fileBufferToString(content))),
+    );
+
+    return this._doHostCall(o, callback);
+  }
+
+  readlink(path: string, callback: Callback<string>): void {
+    const err: NodeJS.ErrnoException = new Error('Not a symlink.');
+    err.code = 'EINVAL';
+    callback(err);
+  }
+
+  statSync(path: string): Stats {
+    if (!this._syncHost) {
+      this._syncHost = new virtualFs.SyncDelegateHost(this._host);
+    }
+
+    const result = this._syncHost.stat(normalize('/' + path));
+    if (result) {
+      return result;
+    } else {
+      return {} as Stats;
+    }
+  }
+  readdirSync(path: string): string[] {
+    if (!this._syncHost) {
+      this._syncHost = new virtualFs.SyncDelegateHost(this._host);
+    }
+
+    return this._syncHost.list(normalize('/' + path));
+  }
+  readFileSync(path: string): string {
+    if (!this._syncHost) {
+      this._syncHost = new virtualFs.SyncDelegateHost(this._host);
+    }
+
+    return virtualFs.fileBufferToString(this._syncHost.read(normalize('/' + path)));
+  }
+  readJsonSync(path: string): string {
+    return JSON.parse(this.readFileSync(path));
+  }
+  readlinkSync(path: string): string {
+    const err: NodeJS.ErrnoException = new Error('Not a symlink.');
+    err.code = 'EINVAL';
+    throw err;
+  }
+
+  purge(_changes?: string[] | string): void {}
```

Comment:
This will need to be implemented in the future.

Reply:
I'm not sure what this should do for our own adapter. For a cache system you can clear the cache, but here I don't thikn there's anything to do.  `purge` is optional on the real typings btw.

Reply:
Let's leave it off and see if there's any problem then.